### ISSUE TITLE
Allow overrides for unresolvable modules

### ISSUE THREAD

**Issue by antoninbas** [uw-labs/lichen#5]
Hi - I am working on replacing https://github.com/mitchellh/golicense with lichen for [Project Antrea](https://github.com/vmware-tanzu/antrea), since golicense is no longer actively maintained and lichen offers me support for Windows binaries.

I am running into one issue during module fetching:
```text
2021/01/29 00:26:16 failed to evaluate licenses: failed to fetch: exit status 1 (output: go: cannot use relative path ../../@ to specify module
```
This is because Antrea uses a `replace` directive pointing to a local directory in one of our go.mod files: https://github.com/vmware-tanzu/antrea/blob/57dcaec0c561875dac6dee9bac72a66b9aefbab6/plugins/octant/go.mod#L14. This is a "trick" that enables us to ensure that plugins are built using the correct version of the "parent" Go module. And it works great for building.

When using golicense, I was able to provide an override (`"../../": "Apache-2.0"`) in the config to handle that case. I'd like to be able to do something similar with lichen, but lichen applies `unresolvableLicense` config overrides after fetching modules and will error out early if a module cannot be fetched.

Would it be possible to provide a new exception type (e.g. `moduleUnfetchable`) in the config that would apply when a module cannot be fetched, instead of returning right away. Thanks!

**Comment by nick-jones** [comment]
Hey,

I've pushed a change to master which I think will cater for this case. I considered the route of adding more exceptions, but I figured in the end it is fairly obvious when a module isn't going to be fetchable. So my change simply avoids attempts to fetch modules where the path looks like a local path. There may be other edge-cases that I've no considered, so I'm definitely not ruling out that idea of making this more configurable - I just figured, keep it simple for now.

With that in mind, `lichen` should now run against what you've linked to. As you'd expect, it'll flag the fact a license cannot be resolved, so either of the following in configuration can help deal with that:


```yaml
override:
  - path: "../../"
    licenses: ["Apache-2.0"]
```

.. or if you don't care to map it to a license, since it's the same codebase:

```yaml
exceptions:
  unresolvableLicense:
    - path: "../../"
```

Let me know if that works for you? If so I'll tag a new version.
